### PR TITLE
Stable changes scw 1.0.13

### PR DESCRIPTION
**Pull request by talhamalik883**

### CODE REVIEW

File: packages/account-abstraction/src/SmartAccountAPI.ts [hunk 1]
```diff
@@ -159,12 +159,6 @@ export class SmartAccountAPI extends BaseAccountAPI {
       }
     }
 
-    if (!maxFeePerGas || !maxPriorityFeePerGas) {
```
Comment (livingrockrises) [inline packages/account-abstraction/src/SmartAccountAPI.ts:162]: why do we need to remove this?

File: packages/account-abstraction/src/calcPreVerificationGas.ts [hunk 2]
```diff
@@ -68,12 +68,13 @@ export function calcPreVerificationGas (userOp: Partial<NotPromise<UserOperation
   } as any
 
   const packed = arrayify(packUserOp(p, false))
+  const lengthInWord = (packed.length + 31) / 32;
   const callDataCost = packed.map(x => x === 0 ? ov.zeroByte : ov.nonZeroByte).reduce((sum, x) => sum + x)
   const ret = Math.round(
     callDataCost +
     ov.fixed / ov.bundleSize +
     ov.perUserOp +
-    ov.perUserOpWord * packed.length
```
Comment (livingrockrises) [inline packages/account-abstraction/src/calcPreVerificationGas.ts:76]: this is from the latest commit on aa packge?

File: packages/account-abstraction/src/BaseAccountAPI.ts [hunk 3]
```diff
@@ -258,7 +258,6 @@ export abstract class BaseAccountAPI {
 
     // Review: doesn't work if wallet is not already deployed.
     const callGasLimit =
-      parseNumber(detailsForUserOp.gasLimit) ??
```
Comment (livingrockrises) [inline packages/account-abstraction/src/BaseAccountAPI.ts:261]: what is the case here at this stage? '// Review: doesn't work if wallet is not already deployed.'

and why this is removed? :  parseNumber(detailsForUserOp.gasLimit) ??

File: packages/account-abstraction/src/ERC4337EthersSigner.ts [hunk 4]
```diff
@@ -51,22 +52,8 @@ export class ERC4337EthersSigner extends Signer {
     const customData: any = transaction.customData
     console.log(customData)
 
-    let gasLimit = 2000000
-
-    if (customData && (customData.isBatchedToMultiSend || !customData.isDeployed)) {
```
Comment (livingrockrises) [inline packages/account-abstraction/src/ERC4337EthersSigner.ts:56]: ok we don't need this anymore?

File: packages/account-abstraction/src/ERC4337EthersSigner.ts [hunk 5]
```diff
@@ -85,7 +72,7 @@ export class ERC4337EthersSigner extends Signer {
         target: transaction.to ?? '',
         data: transaction.data?.toString() ?? '',
         value: transaction.value,
-        gasLimit: isDeployed ? transaction.gasLimit : gasLimit,
+        gasLimit: transaction.gasLimit,
```
Comment (livingrockrises) [inline packages/account-abstraction/src/ERC4337EthersSigner.ts:75]: all the cases work without passing custom gas limit?

File: packages/ethers-lib/src/contracts/SmartWalletFactory/v1.0.0/SmartWalletProxyFactoryEthersContract.ts [hunk 6]
```diff
@@ -16,7 +16,7 @@ class SmartWalletFactoryContractEthers implements SmartWalletFactoryContract {
   }
 
   async isWalletExist(wallet: string): Promise<boolean> {
-    const doesExist = await this.contract.isWalletExist(wallet)
+    const doesExist = await this.contract.isAccountExist(wallet)
```
Comment (livingrockrises) [inline packages/ethers-lib/src/contracts/SmartWalletFactory/v1.0.0/SmartWalletProxyFactoryEthersContract.ts:19]: we will be possibly removing this... watchout for upcoming PR's on c4-remediations branch

File: packages/account-abstraction/src/SmartAccountAPI.ts (hunk 1 above)
Comment (talhamalik883) [reply packages/account-abstraction/src/SmartAccountAPI.ts:162]: there is no need. these variables are already assigned

File: packages/account-abstraction/src/BaseAccountAPI.ts (hunk 3 above)
Comment (talhamalik883) [reply packages/account-abstraction/src/BaseAccountAPI.ts:261]: it works even wallet is not deployed have tested this

File: packages/account-abstraction/src/ERC4337EthersSigner.ts (hunk 4 above)
Comment (talhamalik883) [reply packages/account-abstraction/src/ERC4337EthersSigner.ts:56]: no

File: packages/account-abstraction/src/BaseAccountAPI.ts (hunk 3 above)
Comment (talhamalik883) [reply packages/account-abstraction/src/BaseAccountAPI.ts:261]: naa, it does not work. just check all of my trxs on chain. there was internal failure of adding liquidity to hyphen contracts.

update code to this

```
const callGasLimit = (await this.provider.estimateGas({
          from: this.entryPoint.address,
          to: this.getAccountAddress(),
          data: callData
        }))
        // if wallet is not deployed we need to multiply estimated limit to 3 times to get accurate callGasLimit
        if (!this.isDeployed)
        callGasLimit = callGasLimit.mul(3)
```

File: packages/account-abstraction/src/ERC4337EthersSigner.ts (hunk 5 above)
Comment (talhamalik883) [reply packages/account-abstraction/src/ERC4337EthersSigner.ts:83]: with the above code for callGasLimit estimation. they works fine

File: packages/account-abstraction/src/BaseAccountAPI.ts (hunk 3 above)
Comment (livingrockrises) [reply packages/account-abstraction/src/BaseAccountAPI.ts:261]: why 3 times?

File: packages/account-abstraction/src/BaseAccountAPI.ts [hunk 7]
```diff
@@ -256,14 +256,19 @@ export abstract class BaseAccountAPI {
...
-    const callGasLimit =
-      parseNumber(detailsForUserOp.gasLimit) ??
-      (await this.provider.estimateGas({
-        from: this.entryPoint.address,
-        to: this.getAccountAddress(),
-        data: callData
-      }))
+    let callGasLimit = BigNumber.from(0)
+
+    if (!detailsForUserOp.gasLimit)
```
Comment (livingrockrises) [inline packages/account-abstraction/src/BaseAccountAPI.ts:261]: how is this calculated ad where does it come from?
detailsForUserOp.gasLimit

**Review by livingrockrises — APPROVED** [review]
approved for further tests. version scrapping looks good

**Pull request merged**